src/sage/combinat/colored_permutations.py

```diff
@@ -1235,19 +1247,60 @@ def to_cycles(self, singletons=True, use_min=True, negative_singletons=True):
             cycle = [cycle_first]
             l[i], next_val = False, l[i]
             s = self._colors[i]
+            add_neg = True
             while next_val != cycle_first:
                 cycle.append(s * next_val)
                 s *= self._colors[next_val - 1]
                 l[next_val - 1], next_val = False, l[next_val - 1]
             if s != 1:
                 cycle.extend([-e for e in cycle])
+                add_neg = False
 
             # Add the cycle to the list of cycles
             if singletons or len(cycle) > 1:
                 cycles.append(tuple(cycle))
+                if negative_cycles and add_neg:
+                    cycles.append(tuple([-e for e in cycle]))
 
         return cycles
 
+    def cycle_type(self):
+        r"""
+        Return a pair of partitions of ``len(self)`` corresponding to the
+        signed cycle type of ``self``.
+
+        A *negative cycle* is a cycle `C = (c_0, \ldots, c_{2k-1})` such that
+        `c_0 = c_k`. Any other cycle is positive. For any cycle `C`, we
```

Comment:
I think that this should be `c_0 = -c_k`.  It may be good to include also the alternative definition, that a negative cycle is a cycle `(c_0, \dots, c_m)` that coincides with its negative `(-c_0,\dots,-c_m)`.

It would be better to define the cycle type, and remove the sentence `For any cycle ... (different cycle).`

The definition I know is:

```
A *cycle* is a tuple `C = (c_0, \dots, c_k)` with `\pi(c_i) = c_{i+1}` for `0\ leq i < k` and `\pi(c_k) = c_0`.  If `C` is a cycle, `\bar C = (-c_0, \dots, -c_k)` is also a cycle.  A cycle is * negative*, if `C = \bar C` up to cyclic reordering.  In this case, `k` is necessarily even and the length of `C` is `k/2`.  A *positive cycle* is a pair `C \bar C`, its length is `k`. 

Let `\alpha` be the partition whose parts are the lengths of the positive cycles and let `\beta` be the partition whose parts are the lengths of the negative cycles.  Then `(\alpha, \beta)` is the cycle type of `\pi`.
```

Reply:
I agree with Martin's definition.

Reply:
I have used Martin's definition and wording.